tests/mr_slice.rs

```diff
@@ -157,103 +151,94 @@ mod remote_mr_slice {
+        tokio::time::sleep(Duration::from_secs(1)).await;
     }
 
-    #[tokio::test]
+    // #[tokio::test] has a default single-threaded test runtime that may cause a dead lock.
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
     async fn test() {
-        let addr = "127.0.0.1:19200";
+        let addr = SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), pick_unused_port().unwrap());
         tokio::join!(server(addr), client(addr));
     }
 }
 
 mod remote_mr_slice_overbound {
     mod test1 {
-        use std::{sync::Arc, time::Duration};
-
-        use tokio::time::sleep;
-
         use crate::*;
         const LEN: usize = 4096;
-        async fn server(addr: &str) {
+        async fn server(addr: SocketAddrV4) {
             let rdmalistener = RdmaListener::bind(addr).await.unwrap();
             let rdma = rdmalistener.accept(1, 1, 128).await.unwrap();
             let lmr = rdma.alloc_local_mr(Layout::new::<[u8; LEN]>()).unwrap();
             rdma.send_local_mr(lmr).await.unwrap();
-            sleep(Duration::from_secs(1)).await;
         }
 
-        async fn client(addr: &str) {
-            let rdma = Rdma::connect(addr, 1, 1, 512).await.unwrap();
-            let rmr = Arc::new(rdma.receive_remote_mr().await.unwrap());
+        async fn client(addr: SocketAddrV4) {
+            let rdma = Rdma::connect(addr, 1, 1, 128).await.unwrap();
+            let rmr = rdma.receive_remote_mr().await.unwrap();
             assert_eq!(rmr.length(), LEN);
             #[allow(clippy::reversed_empty_ranges)]
             let _s1 = rmr.get(2..0).unwrap();
         }
 
-        #[tokio::test]
+        // #[tokio::test] has a default single-threaded test runtime that may cause a dead lock.
+        #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
         #[should_panic]
         async fn test() {
-            let addr = "127.0.0.1:19300";
+            let addr = SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), pick_unused_port().unwrap());
             tokio::join!(server(addr), client(addr));
         }
     }
 
     mod test2 {
-        use std::{sync::Arc, time::Duration};
-
-        use tokio::time::sleep;
-
         use crate::*;
         const LEN: usize = 4096;
-        async fn server(addr: &str) {
+        async fn server(addr: SocketAddrV4) {
             let rdmalistener = RdmaListener::bind(addr).await.unwrap();
             let rdma = rdmalistener.accept(1, 1, 128).await.unwrap();
             let lmr = rdma.alloc_local_mr(Layout::new::<[u8; LEN]>()).unwrap();
             rdma.send_local_mr(lmr).await.unwrap();
-            sleep(Duration::from_secs(1)).await;
         }
 
-        async fn client(addr: &str) {
-            let rdma = Rdma::connect(addr, 1, 1, 512).await.unwrap();
-            let rmr = Arc::new(rdma.receive_remote_mr().await.unwrap());
+        async fn client(addr: SocketAddrV4) {
+            let rdma = Rdma::connect(addr, 1, 1, 128).await.unwrap();
```

Comment:
This number is used in many example and test, should be move it into a const value?